### PR TITLE
Add unit tests to common

### PR DESCRIPTION
**Pull request by rlazo**

### CODE REVIEW

File: common/src/main/kotlin/com/google/ai/client/generativeai/common/APIController.kt [hunk 1]
```diff
@@ -79,28 +82,39 @@ class APIController(
...
+      throw GoogleGenerativeAIException.from(e)
     }
-  }
 
-  suspend fun countTokens(request: CountTokensRequest): CountTokensResponse =
+  fun generateContentStream(request: GenerateContentRequest): Flow<GenerateContentResponse> =
     client
-      .post("$DOMAIN/${requestOptions.apiVersion}/$model:countTokens") {
+      .postStream<GenerateContentResponse>(
+        "$DOMAIN/${requestOptions.apiVersion}/$model:streamGenerateContent?alt=sse"
```
[the remark this answers is not in the document]
Comment (rlazo) [reply common/src/main/kotlin/com/google/ai/client/generativeai/common/APIController.kt:100]: That may change depending on other uses of common, will leave it as-is for now. sgty?

File: common/src/test/java/com/google/ai/client/generativeai/common/StreamingSnapshotTests.kt [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+  @Test
+  fun `citation returns correctly when using alternative name`() =
+    goldenStreamingFile("success-citations-altname.txt") {
+      val responses = apiController.generateContentStream(textGenerateContentRequest("prompt"))
+
+      withTimeout(testTimeout) {
+        val responseList = responses.toList()
+        responseList.any {
+          it.candidates?.any { it.citationMetadata?.citationSources?.isNotEmpty() ?: false }
+            ?: false
```
[the remark this answers is not in the document]
Comment (rlazo) [reply common/src/test/java/com/google/ai/client/generativeai/common/StreamingSnapshotTests.kt:159]: It does sound like a great idea... but it triggers a stackoverflow due to recursion

```
Caused by: java.lang.StackOverflowError
	at com.google.ai.client.generativeai.common.util.KotlinKt.any(kotlin.kt:38)
```

File: common/src/main/kotlin/com/google/ai/client/generativeai/common/APIController.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (daymxn) [reply common/src/main/kotlin/com/google/ai/client/generativeai/common/APIController.kt:100]: valid, sgtm

File: common/src/test/java/com/google/ai/client/generativeai/common/StreamingSnapshotTests.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (daymxn) [reply common/src/test/java/com/google/ai/client/generativeai/common/StreamingSnapshotTests.kt:159]: Ahh I see, this should fix it:

```kotlin
internal fun <T> Iterable<T>?.any(predicate: (T) -> Boolean): Boolean =
  this?.firstNotNullOfOrNull(predicate) != null
```

**Pull request merged**